specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ServerDevOpsAudit.json

```diff
@@ -220,6 +220,10 @@
           "description": "Specifies whether DevOps audit events are sent to Azure Monitor. \r\nIn order to send the events to Azure Monitor, specify 'State' as 'Enabled' and 'IsAzureMonitorTargetEnabled' as true.\r\n\r\nWhen using REST API to configure DevOps audit, Diagnostic Settings with 'DevOpsOperationsAudit' diagnostic logs category on the master database should be also created.\r\n\r\nDiagnostic Settings URI format:\r\nPUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Sql/servers/{serverName}/databases/master/providers/microsoft.insights/diagnosticSettings/{settingsName}?api-version=2017-05-01-preview\r\n\r\nFor more information, see [Diagnostic Settings REST API](https://go.microsoft.com/fwlink/?linkid=2033207)\r\nor [Diagnostic Settings PowerShell](https://go.microsoft.com/fwlink/?linkid=2033043)\r\n",
           "type": "boolean"
         },
+        "isManagedIdentityInUse": {
```

Comment:
>isManagedIdentityInUse [](http://example.com/codeflow?start=9&length=22)

Recommend string enum instead of boolean. Enums are better properties for many reasons. In this case, you could do something more like blobStorageAccess: { Default, ManagedIdentity } (or better yet, name the default case). This approach also allows adding new enum values without creating a new API version.

Reply:
This API mimics the existing Auditing API, changing this to an enum would require changes in the auditing API and would require a lot of changes in other parts of the code.

Reply:
the suggestion is very good. 
I guess this parameter is following existing other auditing APIs. the customer might have onboarded the existing bool style auditing APIs for their scripts or workflow.

Reply:
Not blocking ARM signoff on this, just a recommendation.